extensions/stackdriver/stackdriver.cc

```diff
@@ -394,6 +408,9 @@ bool StackdriverRootContext::recordTCP(uint32_t id) {
   getValue({"response", "flags"}, &response_flags);
   if (!peer_found && peer_id != ::Wasm::Common::kMetadataNotFoundValue &&
       response_flags == 0) {
+    logTCPOpenOnTimeout(
```

Comment:
Do we need to add this function call here? Can we extend the condition above to：
```
if (!peer_found && peer_id != ::Wasm::Common::kMetadataNotFoundValue &&
      response_flags == 0 && cur_time - start_time <= timeout)
```
and if it times out, just running through the following logic to record metrics and add logs.

Reply:
we don't plan to record metrics at that time.. just logs... hence having a separate function call here makes it easy to figure out the logic...

Reply:
Should we keep the behavior consistent on logging and metric? I don't see downside on recording connection open on timeout here.

Reply:
@bianpengyuan : thanks fixed.. recording metric on connection open on timeout too.. please review